src/topojson/topojson.ml

```diff
@@ -616,4 +616,54 @@ module Make (J : Intf.Json) = struct
         Geometry.to_json g
 
   let v ?bbox topojson = { bbox; topojson }
+
+  let extract_coordinates = function
+    | `List coordinates ->
+        let rec loop acc = function
+          | [] -> acc
+          | hd :: tl ->
+              let coord =
+                match hd with
+                | `List c -> List.map J.to_float c
+                | _ -> failwith "Invalid coordinate format"
+              in
+              loop (coord :: acc) tl
+        in
+        loop [] coordinates
+    | _ -> failwith "Not a valid LineString or Polygon"
+
+  let extract_lines = function
```

Comment:
This comment applies to pretty much most of the other functions defined here.

I don't think we want to reimplement all of the JSON parsing logic again, GeoJSON does this already for us so we should just uses GeoJSON to extract the coordinates etc. from the geometries (since we are converting from GeoJSON to TopoJSON).

Reply:
e.g. 

https://github.com/geocaml/ocaml-geojson/blob/d45db9166a050da2e61ff3f04856c4788c13edc8/src/geojson/geojson_intf.ml#L124